src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -88,48 +86,62 @@ internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
 
         private void WriteLocations(Location location, IReadOnlyList<Location> additionalLocations)
         {
-            _writer.WriteArrayStart("locations");
+            if (location.SourceTree != null)
+            {
+                _writer.WriteArrayStart("locations");
+                _writer.WriteObjectStart(); // location
+                _writer.WriteKey("analysisTarget");
```

Comment:
not sure I agree. the 'compilation' as Nick says, is the analysis target, not an individual source file. imagine that Roslyn fires a result against 'SharedFile.cs' which is compiled into multiple projects. what exists in the Roslyn output to clarify that it is a specific assembly's use of this file? this exactly conforms to the C++ case you mention (shared header file) yet in this case, you propose to put the shared file in analysisTarget.

For Roslyn, I tend to think that the project file may be the analysisTarget.